Vendor in CDI 1.4.1

dylanzr opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:
1. Follow the instructions for writing a release note from k8s: https://git.k8s.io/community/contributors/guide/release-notes.md
-->

**What this PR does / why we need it**:
Updates CDI and testing manifest. Verified that DataVolumeTemplates now support blank volumes.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:
<!--  Write your release note:
1. Enter your extended release note in the below block. If the PR requires additional action from users switching to the new release, include the string "action required".
2. If no release note is required, just write "NONE".
-->
```release-note
NONE
```

ovirt-infra commented:
Hello contributor, thanks for submitting a PR for this project! 

I am the bot who triggers "standard-CI" builds for this project.
As a security measure, I will not run automated tests on PRs that are not from white-listed contributors.

In order to allow automated tests to run, please ask one of the project maintainers to review the code and then do one of the following:

1. Type `ci test please` on this PR to trigger automated tests for it.
2. Type `ci add to whitelist` on this PR to trigger automated tests for it and also add you to the contributor white-list so that your future PRs will be tested automatically.
3. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the project organization, so you'll be able to run tests for all the organization's projects.

slintes commented:
ci test please

mmazur commented:
`validation:` in the manifest fails with: `error validating data: [unknown object type "nil" in CustomResourceDefinition.spec.validation.openAPIV3Schema.properties.spec.properties.source.properties.blank, unknown object type "nil" in CustomResourceDefinition.spec.validation.openAPIV3Schema.properties.spec.properties.source.properties.upload];`.

Either needs to be removed for the time being from the patch or upstream CDI consulted on how to fix this properly. Ping @aglitke @awels

mhenriks commented:
@dylanzr @mmazur See this pr for validation fix:   https://github.com/kubevirt/containerized-data-importer/pull/557

dylanzr commented:
@mhenriks Thanks, fix has been pushed.

ci test please

mhenriks commented:
ci test please

mmazur commented:
So… what needs to happen for this to get merged?

cynepco3hahue commented:
ci test please

cynepco3hahue commented:
@mmazur let's wait for additional CI run and if it will not have any CDI related issues I will merge it

mmazur commented:
@cynepco3hahue ok, so what happens now with all the failed tests? I clicked on a random one and am seeing stuff like `[Fail] Networking VirtualMachineInstance with custom PCI address [It] should configure custom Pci address`, which is very much unrelated to anything CDI. Does someone just decide to ignore the tests and merge anyway maybe?

mmazur commented:
Ah, now I get it. Darn, some of the tests are cdi–related and they do fail.

dylanzr commented:
I don't see this specifically failing, but I think this must be the http import failing. I'll have some time to look into this later today.

dylanzr commented:
I looked into this a bit more last night. I was having some issues with the functest as there's not a 'local' storage class in my cluster, but with a dynamic provisioner I could get the tests to run and pass with CDI 1.3. However, with 1.4.1 the 'Starting a VirtualMachineInstance with a DataVolume as a volume source' test seems to error out immediately when the first warning event for an unbound PVC happens. I've confirmed that the HTTP import actually does work, so I'll be digging into the test later today, as time allows, to see why the test isn't waiting for the eventual success.

slintes commented:
@dylanzr without looking into the details, as a hint: if Warning events are the problem, you might want to check if this method is called on the event watcher, it activates fail on warning: https://github.com/kubevirt/kubevirt/blob/ff476cae19647486941ad63b92597839c02217cb/tests/utils.go#L204

mmazur commented:
This should be updated to most recent master for CDI, as 1.4.1 as released has problems in api definitions which confuse clients: https://github.com/kubevirt/containerized-data-importer/commit/80f8dcf8d83c57b6a24e93819fe60518be73a8bc#diff-fdfac9813673596faeda07fdce9207e2L434

dylanzr commented:
@mmazur, It might be a few days before I could look at this again. I've still been unable to get the functests to behave, and I'm not sure if it's due to something I've introduced by creating a "local" storageclass (rook ceph flexvolume provisioner) that behaves differently than expected, but even in that environment 1.3 does pass. Essentially, the warnings for things like PVC not being bound or the pod waiting for the PVC are triggering errors for the VMI tests.

mmazur commented:
@dylanzr no hurry, just wanted to make sure that once this is ready for merging, it's a version that will work correctly, as the 1.4.1 without Michael's patch has issues.

Btw, @mhenriks, I believe the convention for both name and singularName is to be all lowercase (regarding https://github.com/kubevirt/containerized-data-importer/blob/master/pkg/apiserver/apiserver.go#L443 ), otherwise you're depending on client libs to make sure to do it, which [might cause issues](https://github.com/openshift/openshift-restclient-python/pull/250).

davidvossel commented on manifests/testing/cdi-v1.4.1.yaml.in:
```diff
@@ -161,6 +222,39 @@ rules:
+kind: RoleBinding
+metadata:
+  name: cdi-extension-apiserver-authentication
+  namespace: kube-system
```
shouldn't this be {{.CDINamespace}} ?

davidvossel commented on manifests/testing/cdi-v1.4.1.yaml.in:
```diff
@@ -161,6 +222,39 @@ rules:
+kind: Role
+metadata:
+  name: cdi-extension-apiserver-authentication
+  namespace: kube-system
```
CDINamespace?

davidvossel commented on manifests/testing/cdi-v1.4.1.yaml.in:
```diff
@@ -47,7 +50,6 @@ apiVersion: rbac.authorization.k8s.io/v1beta1
 kind: ClusterRoleBinding
 metadata:
   name: cdi-sa
-  namespace: {{.CDINamespace}}
```
should this be removed?

dylanzr replied:
ClusterRole and ClusterRoleBinding aren't namespaced.

dylanzr commented:
@davidvossel I'll investigate those changes after rebasing today.

dylanzr commented:
About to push up the rebase and modifications mentioned above, but here's more detail into the issues I'm encountering.


<details><summary>CDI 1.4.1 - Starting a VirtualMachineInstance with a DataVolume as a volume source - environment</summary>
<p>

```bash

[root@node1-mia1 cluster]# kubectl -n kubevirt-test-default get events --sort-by='{.lastTimestamp}' 
LAST SEEN   TYPE      REASON                       KIND                     MESSAGE
2m30s       Normal    ExternalProvisioning         PersistentVolumeClaim    waiting for a volume to be created, either by external provisioner "ceph.rook.io/block" or manually created by system administrator
2m29s       Normal    Scheduled                    Pod                      Successfully assigned kubevirt-test-default/importer-test-datavolume-fk2t45gkznqq-m8782 to node1-mia1.domain.com
2m29s       Warning   FailedDataVolumeImport       VirtualMachineInstance   DataVolume test-datavolume-fk2t45gkznqq failed
2m29s       Warning   FailedScheduling             Pod                      pod has unbound immediate PersistentVolumeClaims (repeated 3 times)
2m29s       Normal    ProvisioningSucceeded        PersistentVolumeClaim    Successfully provisioned volume pvc-bf08e733-03d5-11e9-909c-f4e9d4724596
2m29s       Normal    Provisioning                 PersistentVolumeClaim    External provisioner is provisioning volume for claim "kubevirt-test-default/test-datavolume-fk2t45gkznqq"
2m26s       Warning   Failed                       Pod                      Error: cannot find volume "cdi-data-vol" to mount into container "importer"
2m26s       Normal    Pulled                       Pod                      Container image "kubevirt/cdi-importer:v1.4.1" already present on machine
2m26s       Normal    Scheduled                    Pod                      Successfully assigned kubevirt-test-default/importer-test-datavolume-m9gk7b4hg92h-nm476 to node2-mia1.domain.com
2m26s       Warning   FailedScheduling             Pod                      pod has unbound immediate PersistentVolumeClaims (repeated 3 times)
2m26s       Normal    ImportScheduled              DataVolume               Import into test-datavolume-fk2t45gkznqq scheduled
2m26s       Normal    SuccessfulDataVolumeCreate   VirtualMachine           Created DataVolume test-datavolume-m9gk7b4hg92h
2m26s       Normal    ImportScheduled              DataVolume               Import into test-datavolume-m9gk7b4hg92h scheduled
2m26s       Normal    ProvisioningSucceeded        PersistentVolumeClaim    Successfully provisioned volume pvc-c1bbc45e-03d5-11e9-909c-f4e9d4724596
2m26s       Normal    Provisioning                 PersistentVolumeClaim    External provisioner is provisioning volume for claim "kubevirt-test-default/test-datavolume-m9gk7b4hg92h"
2m24s       Warning   ImportFailed                 DataVolume               Failed to import into PVC test-datavolume-m9gk7b4hg92h
2m22s       Normal    Pulled                       Pod                      Container image "kubevirt/cdi-importer:v1.4.1" already present on machine
2m22s       Normal    Created                      Pod                      Created container
2m22s       Normal    Started                      Pod                      Started container
2m21s       Normal    ImportInProgress             DataVolume               Import into test-datavolume-m9gk7b4hg92h in progress
2m21s       Normal    Scheduled                    Pod                      Successfully assigned kubevirt-test-default/importer-test-datavolume-fk2t45gkznqq-2sdnn to node3-mia1.domain.com
2m21s       Warning   ImportFailed                 DataVolume               Failed to import into PVC test-datavolume-fk2t45gkznqq
2m19s       Normal    Synced                       DataVolume               DataVolume synced successfully
2m19s       Normal    ImportSucceeded              DataVolume               Successfully imported into PVC test-datavolume-m9gk7b4hg92h
2m18s       Normal    Pulled                       Pod                      Container image "kubevirt/cdi-importer:v1.4.1" already present on machine
2m17s       Normal    Started                      Pod                      Started container
2m17s       Normal    ImportInProgress             DataVolume               Import into test-datavolume-fk2t45gkznqq in progress
2m17s       Normal    Created                      Pod                      Created container
2m16s       Normal    ImportSucceeded              DataVolume               Successfully imported into PVC test-datavolume-fk2t45gkznqq
2m16s       Normal    Synced                       DataVolume               DataVolume synced successfully
2m6s        Normal    SuccessfulCreate             VirtualMachine           Created virtual machine: testvmivx2cgtcgppwmtsz97j2sfs9tspnvvjlsx7kl24vmhshl7h45
2m6s        Normal    SuccessfulCreate             VirtualMachineInstance   Created virtual machine pod virt-launcher-testvmivx2cgtcgppwmtsz97j2sfs9tspnvvjlsx7kl29l4sq
2m6s        Normal    Scheduled                    Pod                      Successfully assigned kubevirt-test-default/virt-launcher-testvmivx2cgtcgppwmtsz97j2sfs9tspnvvjlsx7kl29l4sq to node3-mia1.domain.com
2m2s        Normal    Started                      Pod                      Started container
2m2s        Normal    Created                      Pod                      Created container
2m2s        Normal    Pulled                       Pod                      Container image "private.registry:443/kubevirt/virt-launcher:master4" already present on machine
112s        Warning   Unhealthy                    Pod                      Readiness probe failed: cat: /var/run/kubevirt-infra/healthy: No such file or directory
110s        Normal    SuccessfulDelete             VirtualMachineInstance   Deleted virtual machine pod virt-launcher-testvmivx2cgtcgppwmtsz97j2sfs9tspnvvjlsx7kl29l4sq
110s        Normal    Created                      VirtualMachineInstance   VirtualMachineInstance defined.
110s        Normal    Deleted                      VirtualMachineInstance   VirtualMachineInstance stopping
110s        Normal    SuccessfulDelete             VirtualMachine           Deleted virtual machine: cdccdd5b-03d5-11e9-909c-f4e9d4724596
110s        Normal    Started                      VirtualMachineInstance   VirtualMachineInstance started.
109s        Normal    Deleted                      VirtualMachineInstance   Signaled Deletion
109s        Warning   Stopped                      VirtualMachineInstance   The VirtualMachineInstance crashed.
94s         Normal    Killing                      Pod                      Killing container with id cri-o://compute:Need to kill Pod
94s         Warning   Unhealthy                    Pod                      Readiness probe errored: rpc error: code = Unknown desc = container is not created or running
39s         Normal    ExternalProvisioning         PersistentVolumeClaim    waiting for a volume to be created, either by external provisioner "ceph.rook.io/block" or manually created by system administrator
24s         Normal    ExternalProvisioning         PersistentVolumeClaim    waiting for a volume to be created, either by external provisioner "ceph.rook.io/block" or manually created by system administrator
7s          Warning   ProvisioningFailed           PersistentVolumeClaim    Failed to provision volume with StorageClass "local": claim Selector is not supported
7s          Normal    Provisioning                 PersistentVolumeClaim    External provisioner is provisioning volume for claim "kubevirt-test-default/disk-windows"
7s          Warning   ProvisioningFailed           PersistentVolumeClaim    Failed to provision volume with StorageClass "local": claim Selector is not supported
7s          Normal    Provisioning                 PersistentVolumeClaim    External provisioner is provisioning volume for claim "kubevirt-test-default/disk-rhel"

[root@node1-mia1 cluster]# kubectl -n kubevirt-test-default describe datavolumes test-datavolume-fk2t45gkznqq 
Name:         test-datavolume-fk2t45gkznqq
Namespace:    kubevirt-test-default
Labels:       <none>
Annotations:  <none>
API Version:  cdi.kubevirt.io/v1alpha1
Kind:         DataVolume
Metadata:
  Creation Timestamp:  2018-12-19T21:33:37Z
  Generation:          1
  Resource Version:    7986349
  Self Link:           /apis/cdi.kubevirt.io/v1alpha1/namespaces/kubevirt-test-default/datavolumes/test-datavolume-fk2t45gkznqq
  UID:                 bf05b95c-03d5-11e9-909c-f4e9d4724596
Spec:
  Pvc:
    Access Modes:
      ReadWriteOnce
    Resources:
      Requests:
        Storage:         2Gi
    Storage Class Name:  local
  Source:
    Http:
      URL:  http://cdi-http-import-server.kubevirt/images/alpine.iso
Status:
  Phase:  Succeeded
Events:
  Type     Reason            Age                 From                   Message
  ----     ------            ----                ----                   -------
  Normal   ImportScheduled   75s (x2 over 78s)   datavolume-controller  Import into test-datavolume-fk2t45gkznqq scheduled
  Warning  ImportFailed      70s (x2 over 78s)   datavolume-controller  Failed to import into PVC test-datavolume-fk2t45gkznqq
  Normal   ImportInProgress  66s                 datavolume-controller  Import into test-datavolume-fk2t45gkznqq in progress
  Normal   Synced            65s (x18 over 80s)  datavolume-controller  DataVolume synced successfully
  Normal   ImportSucceeded   65s                 datavolume-controller  Successfully imported into PVC test-datavolume-fk2t45gkznqq

```

</p>
</details>

<details><summary>CDI 1.4.1 - Starting a VirtualMachineInstance with a DataVolume as a volume source - output </summary>
<p>

```bash
I1219 21:33:22.619778       1 uploadproxy.go:47] Note: increase the -v level in the api deployment for more detailed logging, eg. -v=2 or -v=3
W1219 21:33:22.621344       1 client_config.go:552] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
apiVersion: kubevirt.io/v1alpha2
kind: VirtualMachineInstance
metadata:
  creationTimestamp: 2018-12-19T21:33:37Z
  generation: 1
  name: testvmi7hb76jxndrh9bjkxfxf2wnpm9nr485v6r4wrf87jztqdn886
  namespace: kubevirt-test-default
  resourceVersion: "7986196"
  selfLink: /apis/kubevirt.io/v1alpha2/namespaces/kubevirt-test-default/virtualmachineinstances/testvmi7hb76jxndrh9bjkxfxf2wnpm9nr485v6r4wrf87jztqdn886
  uid: bf08de0f-03d5-11e9-909c-f4e9d4724596
spec:
  domain:
    devices:
      disks:
      - disk:
          bus: virtio
        name: disk0
        volumeName: disk0
      interfaces:
      - bridge: {}
        name: default
    features:
      acpi:
        enabled: true
    firmware:
      uuid: 90d510e7-c9e8-4f2d-b574-b4170f6ed385
    machine:
      type: q35
    resources:
      requests:
        memory: 512M
  networks:
  - name: default
    pod: {}
  terminationGracePeriodSeconds: 0
  volumes:
  - dataVolume:
      name: test-datavolume-fk2t45gkznqq
    name: disk0
status:
  phase: Failed


Pod name: importer-test-datavolume-fk2t45gkznqq-m8782    Pod phase: Pending

apiVersion: kubevirt.io/v1alpha2
kind: VirtualMachineInstance
metadata:
  creationTimestamp: 2018-12-19T21:33:37Z
  generation: 1
  name: testvmi7hb76jxndrh9bjkxfxf2wnpm9nr485v6r4wrf87jztqdn886
  namespace: kubevirt-test-default
  resourceVersion: "7986196"
  selfLink: /apis/kubevirt.io/v1alpha2/namespaces/kubevirt-test-default/virtualmachineinstances/testvmi7hb76jxndrh9bjkxfxf2wnpm9nr485v6r4wrf87jztqdn886
  uid: bf08de0f-03d5-11e9-909c-f4e9d4724596
spec:
  domain:
    devices:
      disks:
      - disk:
          bus: virtio
        name: disk0
        volumeName: disk0
      interfaces:
      - bridge: {}
        name: default
    features:
      acpi:
        enabled: true
    firmware:
      uuid: 90d510e7-c9e8-4f2d-b574-b4170f6ed385
    machine:
      type: q35
    resources:
      requests:
        memory: 512M
  networks:
  - name: default
    pod: {}
  terminationGracePeriodSeconds: 0
  volumes:
  - dataVolume:
      name: test-datavolume-fk2t45gkznqq
    name: disk0
status:
  phase: Failed

STEP: Starting and stopping the VirtualMachineInstance a number of times
STEP: Starting a VirtualMachineInstance with DataVolume
STEP: Waiting until the VirtualMachineInstance will start
{"component":"tests","kind":"VirtualMachineInstance","level":"error","msg":"Event(v1.ObjectReference{Kind:\"VirtualMachineInstance\", Namespace:\"kubevirt-test-default\", Name:\"testvmi7hb76jxndrh9bjkxfxf2wnpm9nr485v6r4wrf87jztqdn886\", UID:\"bf08de0f-03d5-11e9-909c-f4e9d4724596\", APIVersion:\"kubevirt.io/v1alpha2\", ResourceVersion:\"7986159\", FieldPath:\"\"}): type: 'Warning' reason: 'FailedDataVolumeImport' DataVolume test-datavolume-fk2t45gkznqq failed","name":"testvmi7hb76jxndrh9bjkxfxf2wnpm9nr485v6r4wrf87jztqdn886","namespace":"kubevirt-test-default","pos":"utils.go:284","reason":"unexpected warning event received","timestamp":"2018-12-19T21:33:39.524827Z","uid":"bf08de0f-03d5-11e9-909c-f4e9d4724596"}

------------------------------
• Failure [4.724 seconds]
DataVolume Integration
/root/go/src/kubevirt.io/kubevirt/tests/datavolume_test.go:37
  Starting a VirtualMachineInstance with a DataVolume as a volume source
  /root/go/src/kubevirt.io/kubevirt/tests/datavolume_test.go:67
    using Alpine import
    /root/go/src/kubevirt.io/kubevirt/tests/datavolume_test.go:68
      should be successfully started and stopped multiple times [It]
      /root/go/src/kubevirt.io/kubevirt/tests/datavolume_test.go:69

      Unexpected Warning event received: testvmi7hb76jxndrh9bjkxfxf2wnpm9nr485v6r4wrf87jztqdn886,bf08de0f-03d5-11e9-909c-f4e9d4724596: DataVolume test-datavolume-fk2t45gkznqq failed
      Expected
          <string>: Warning
      not to equal
          <string>: Warning

      /root/go/src/kubevirt.io/kubevirt/tests/utils.go:318
------------------------------
```

</p>
</details>

<details><summary>CDI 1.4.1 - cluster/examples/vm-alpine-datavolume.yaml</summary>
<p>

```bash
[root@node1-mia1 cluster]# kubectl -n default get events --sort-by='{.lastTimestamp}' 
LAST SEEN   TYPE      REASON               KIND               MESSAGE
2m8s   Normal    ProvisioningSucceeded        PersistentVolumeClaim    Successfully provisioned volume pvc-61de7a73-03d6-11e9-909c-f4e9d4724596
2m8s   Normal    ImportScheduled              DataVolume               Import into alpine-dv scheduled
2m8s   Warning   ImportFailed                 DataVolume               Failed to import into PVC alpine-dv
2m8s   Normal    Scheduled                    Pod                      Successfully assigned default/importer-alpine-dv-jc2bj to node2-mia1.domain.com
2m8s   Normal    Provisioning                 PersistentVolumeClaim    External provisioner is provisioning volume for claim "default/alpine-dv"
2m8s   Warning   FailedScheduling             Pod                      pod has unbound immediate PersistentVolumeClaims (repeated 3 times)
2m8s   Normal    SuccessfulDataVolumeCreate   VirtualMachine           Created DataVolume alpine-dv
2m5s   Normal    Pulled                       Pod                      Container image "kubevirt/cdi-importer:v1.4.1" already present on machine
2m5s   Normal    Created                      Pod                      Created container
2m5s   Normal    Started                      Pod                      Started container
2m4s   Normal    ImportInProgress             DataVolume               Import into alpine-dv in progress
2m3s   Normal    ImportSucceeded              DataVolume               Successfully imported into PVC alpine-dv
2m3s   Normal    Synced                       DataVolume               DataVolume synced successfully
47s    Normal    SuccessfulCreate             VirtualMachineInstance   Created virtual machine pod virt-launcher-vm-alpine-datavolume-6b5tw
47s    Normal    Scheduled                    Pod                      Successfully assigned default/virt-launcher-vm-alpine-datavolume-6b5tw to node3-mia1.domain.com
47s    Normal    SuccessfulCreate             VirtualMachine           Created virtual machine: vm-alpine-datavolume
43s    Normal    Created                      Pod                      Created container
43s    Normal    Started                      Pod                      Started container
43s    Normal    Pulled                       Pod                      Container image "private.registry:443/kubevirt/virt-launcher:master4" already present on machine
34s    Warning   Unhealthy                    Pod                      Readiness probe failed: cat: /var/run/kubevirt-infra/healthy: No such file or directory
32s    Normal    Created                      VirtualMachineInstance   VirtualMachineInstance defined.
32s    Normal    Started                      VirtualMachineInstance   VirtualMachineInstance started.
```
</p>
</details>

<details><summary>CDI 1.3 - Starting a VirtualMachineInstance with a DataVolume as a volume source - environment</summary>
<p>

```bash

[root@node1-mia1 kubevirt]# kubectl -n kubevirt-test-default get events --sort-by='{.lastTimestamp}' 
LAST SEEN   TYPE      REASON                       KIND                     MESSAGE
2m31s       Normal    Provisioning                 PersistentVolumeClaim    External provisioner is provisioning volume for claim "kubevirt-test-default/test-datavolume-ng6n4cxw9xnm"
2m31s       Normal    ExternalProvisioning         PersistentVolumeClaim    waiting for a volume to be created, either by external provisioner "ceph.rook.io/block" or manually created by system administrator
2m30s       Normal    ImportScheduled              DataVolume               Import into test-datavolume-ng6n4cxw9xnm scheduled
2m30s       Normal    ProvisioningSucceeded        PersistentVolumeClaim    Successfully provisioned volume pvc-082f5797-03de-11e9-8f9f-f4e9d47243f8
2m30s       Normal    Scheduled                    Pod                      Successfully assigned kubevirt-test-default/importer-test-datavolume-ng6n4cxw9xnm-2bxps to node3-mia1.domain.com
2m30s       Warning   FailedScheduling             Pod                      pod has unbound immediate PersistentVolumeClaims (repeated 3 times)
2m27s       Normal    Pulled                       Pod                      Container image "kubevirt/cdi-importer:v1.3.0" already present on machine
2m27s       Normal    Started                      Pod                      Started container
2m27s       Normal    Created                      Pod                      Created container
2m26s       Normal    ImportSucceded               DataVolume               Successfully imported into PVC test-datavolume-ng6n4cxw9xnm
2m26s       Normal    Scheduled                    Pod                      Successfully assigned kubevirt-test-default/virt-launcher-testvmijmprn762h2sx2z8hbszkldrv6bdwwt8csvbms8rc4z to node1-mia1.domain.com
2m26s       Normal    Synced                       DataVolume               DataVolume synced successfully
2m26s       Normal    SuccessfulCreate             VirtualMachineInstance   Created virtual machine pod virt-launcher-testvmijmprn762h2sx2z8hbszkldrv6bdwwt8csvbms8rc4z
2m23s       Normal    Pulling                      Pod                      pulling image "private.registry/kubevirt/virt-launcher:cdi13"
2m18s       Normal    Created                      Pod                      Created container
2m18s       Normal    Started                      Pod                      Started container
2m18s       Normal    Pulled                       Pod                      Successfully pulled image "private.registry/kubevirt/virt-launcher:cdi13"
2m8s        Warning   Unhealthy                    Pod                      Readiness probe failed: cat: /var/run/kubevirt-infra/healthy: No such file or directory
2m6s        Warning   Stopped                      VirtualMachineInstance   The VirtualMachineInstance crashed.
2m6s        Normal    Deleted                      VirtualMachineInstance   Signaled Deletion
2m6s        Normal    Deleted                      VirtualMachineInstance   VirtualMachineInstance stopping
2m6s        Normal    SuccessfulDelete             VirtualMachineInstance   Deleted virtual machine pod virt-launcher-testvmijmprn762h2sx2z8hbszkldrv6bdwwt8csvbms8rc4z
2m6s        Normal    Started                      VirtualMachineInstance   VirtualMachineInstance started.
2m6s        Normal    Created                      VirtualMachineInstance   VirtualMachineInstance defined.
110s        Warning   Unhealthy                    Pod                      Readiness probe errored: rpc error: code = Unknown desc = container is not created or running
110s        Normal    Killing                      Pod                      Killing container with id cri-o://compute:Need to kill Pod
99s         Normal    Scheduled                    Pod                      Successfully assigned kubevirt-test-default/virt-launcher-testvmijmprn762h2sx2z8hbszkldrv6bdwwt8csvbmsbmwgj to node1-mia1.domain.com
99s         Normal    SuccessfulCreate             VirtualMachineInstance   Created virtual machine pod virt-launcher-testvmijmprn762h2sx2z8hbszkldrv6bdwwt8csvbmsbmwgj
96s         Normal    Created                      Pod                      Created container
96s         Normal    Started                      Pod                      Started container
96s         Normal    Pulled                       Pod                      Container image "private.registry/kubevirt/virt-launcher:cdi13" already present on machine
86s         Warning   Unhealthy                    Pod                      Readiness probe failed: cat: /var/run/kubevirt-infra/healthy: No such file or directory
84s         Normal    Started                      VirtualMachineInstance   VirtualMachineInstance started.
84s         Normal    Created                      VirtualMachineInstance   VirtualMachineInstance defined.
80s         Warning   Stopped                      VirtualMachineInstance   The VirtualMachineInstance crashed.
80s         Normal    Deleted                      VirtualMachineInstance   Signaled Deletion
80s         Normal    Deleted                      VirtualMachineInstance   VirtualMachineInstance stopping
80s         Normal    SuccessfulDelete             VirtualMachineInstance   Deleted virtual machine pod virt-launcher-testvmijmprn762h2sx2z8hbszkldrv6bdwwt8csvbmsbmwgj
80s         Normal    Killing                      Pod                      Killing container with id cri-o://compute:Need to kill Pod
80s         Warning   Unhealthy                    Pod                      Readiness probe errored: rpc error: code = Unknown desc = container is not created or running
68s         Normal    Provisioning                 PersistentVolumeClaim    External provisioner is provisioning volume for claim "kubevirt-test-default/test-datavolume-6md7pbn94gn8"
68s         Normal    SuccessfulDataVolumeCreate   VirtualMachine           Created DataVolume test-datavolume-6md7pbn94gn8
67s         Normal    ProvisioningSucceeded        PersistentVolumeClaim    Successfully provisioned volume pvc-3a6a5c72-03de-11e9-8f9f-f4e9d47243f8
67s         Normal    ImportScheduled              DataVolume               Import into test-datavolume-6md7pbn94gn8 scheduled
67s         Warning   FailedScheduling             Pod                      pod has unbound immediate PersistentVolumeClaims (repeated 3 times)
67s         Normal    Scheduled                    Pod                      Successfully assigned kubevirt-test-default/importer-test-datavolume-6md7pbn94gn8-dfvfz to node3-mia1.domain.com
63s         Normal    SuccessfulCreate             VirtualMachineInstance   Created virtual machine pod virt-launcher-testvmimndp5krccg7gr8gkr4gl86wdvc6d2rdhfshgjmfd5q
63s         Normal    SuccessfulCreate             VirtualMachine           Created virtual machine: testvmimndp5krccg7gr8gkr4gl86wdvc6d2rdhfshgjvxbzkl9gs9h
63s         Normal    Synced                       DataVolume               DataVolume synced successfully
63s         Normal    Created                      Pod                      Created container
63s         Normal    Pulled                       Pod                      Container image "kubevirt/cdi-importer:v1.3.0" already present on machine
63s         Normal    Started                      Pod                      Started container
63s         Normal    ImportSucceded               DataVolume               Successfully imported into PVC test-datavolume-6md7pbn94gn8
62s         Normal    Scheduled                    Pod                      Successfully assigned kubevirt-test-default/virt-launcher-testvmimndp5krccg7gr8gkr4gl86wdvc6d2rdhfshgjmfd5q to node1-mia1.domain.com
61s         Normal    SuccessfulDelete             VirtualMachineInstance   Deleted virtual machine pod virt-launcher-testvmimndp5krccg7gr8gkr4gl86wdvc6d2rdhfshgjmfd5q
60s         Warning   ProvisioningFailed           PersistentVolumeClaim    Failed to provision volume with StorageClass "local": claim Selector is not supported
60s         Normal    Provisioning                 PersistentVolumeClaim    External provisioner is provisioning volume for claim "kubevirt-test-default/disk-windows"
60s         Normal    ExternalProvisioning         PersistentVolumeClaim    waiting for a volume to be created, either by external provisioner "ceph.rook.io/block" or manually created by system administrator
60s         Warning   ProvisioningFailed           PersistentVolumeClaim    storageclass.storage.k8s.io "host-path" not found
60s         Normal    ExternalProvisioning         PersistentVolumeClaim    waiting for a volume to be created, either by external provisioner "ceph.rook.io/block" or manually created by system administrator
60s         Warning   ProvisioningFailed           PersistentVolumeClaim    Failed to provision volume with StorageClass "local": claim Selector is not supported
60s         Normal    Provisioning                 PersistentVolumeClaim    External provisioner is provisioning volume for claim "kubevirt-test-default/disk-rhel"
56s         Warning   FailedCreatePodSandBox       Pod                      Failed create pod sandbox: rpc error: code = Unknown desc = failed to get network status for pod sandbox k8s_virt-launcher-testvmimndp5krccg7gr8gkr4gl86wdvc6d2rdhfshgjmfd5q_kubevirt-test-default_3d66ad1f-03de-11e9-909c-f4e9d4724596_0(feb3585df4618fa04990f5f6dfbbe480a8a9829c282705fa43481946047112ae): Unexpected command output nsenter: cannot open /proc/24200/ns/net: No such file or directory
 with error: exit status 1

 ```

</p>
</details>

# 

The DV import does eventually work, and the VM will launch, however, there's a `VirtualMachineInstance` warning for `FailedDataVolumeImport` that isn't present in the older version. This warning is probably happening when the import pod is waiting for the PVC to become available. I think this is actually failing somewhere in  waitForVMIStart(). Still working through it.

cynepco3hahue commented:
@dylanzr You can use `WaitForSuccessfulVMIStartIgnoreWarnings` to ignore warnings.

mhenriks commented:
ci test please

dylanzr commented:
@cynepco3hahue I've done that, but the next check fails the 'VMI unexpectedly stopped' immediately after the 'ignoreWarnings' conditional. I'll be looking at this more closely throughout the day today. I'm also going to be looking into what the kubevirt ci environments are doing for the local storage class, as it might be related to something specific to my dynamic provisioning.

dylanzr commented:
I've added a step in the test that catches the DV status after vmi creation and ensures it reaches the succeeded phase. I'll get it cleaned up and pushed later tonight or in the morning

dylanzr commented:
Ok, I've rebased this again and added a function to check the status of the DataVolume before creating the VMI. This appears to work now. The following output has some extra debugging, but demonstrates the functionality.

<details><summary>successful test output</summary>
<p>

```

SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSdatavolume: test-datavolume-thcdqz4fp6c7  phase: 
datavolume: test-datavolume-thcdqz4fp6c7  phase: Pending
datavolume: test-datavolume-thcdqz4fp6c7  phase: Failed
datavolume: test-datavolume-thcdqz4fp6c7  phase: Failed
datavolume: test-datavolume-thcdqz4fp6c7  phase: Failed
datavolume: test-datavolume-thcdqz4fp6c7  phase: Failed
datavolume: test-datavolume-thcdqz4fp6c7  phase: Failed
datavolume: test-datavolume-thcdqz4fp6c7  phase: ImportInProgress
datavolume: test-datavolume-thcdqz4fp6c7  phase: ImportInProgress
datavolume: test-datavolume-thcdqz4fp6c7  phase: ImportInProgress
datavolume: test-datavolume-thcdqz4fp6c7  phase: Succeeded
vmi: testvmi4p7jpmzsmtpsjmj55tq2cfzvdnsn5dfwfds5tftpbfvxkl5j  phase: 
vmi: testvmi4p7jpmzsmtpsjmj55tq2cfzvdnsn5dfwfds5tftpbfvxkl5j  phase: Scheduling
vmi: testvmi4p7jpmzsmtpsjmj55tq2cfzvdnsn5dfwfds5tftpbfvxkl5j  phase: Scheduling
vmi: testvmi4p7jpmzsmtpsjmj55tq2cfzvdnsn5dfwfds5tftpbfvxkl5j  phase: Scheduling
vmi: testvmi4p7jpmzsmtpsjmj55tq2cfzvdnsn5dfwfds5tftpbfvxkl5j  phase: Scheduling
vmi: testvmi4p7jpmzsmtpsjmj55tq2cfzvdnsn5dfwfds5tftpbfvxkl5j  phase: Scheduling
vmi: testvmi4p7jpmzsmtpsjmj55tq2cfzvdnsn5dfwfds5tftpbfvxkl5j  phase: Scheduling
vmi: testvmi4p7jpmzsmtpsjmj55tq2cfzvdnsn5dfwfds5tftpbfvxkl5j  phase: Scheduling
vmi: testvmi4p7jpmzsmtpsjmj55tq2cfzvdnsn5dfwfds5tftpbfvxkl5j  phase: Scheduling
vmi: testvmi4p7jpmzsmtpsjmj55tq2cfzvdnsn5dfwfds5tftpbfvxkl5j  phase: Scheduling
vmi: testvmi4p7jpmzsmtpsjmj55tq2cfzvdnsn5dfwfds5tftpbfvxkl5j  phase: Scheduling
vmi: testvmi4p7jpmzsmtpsjmj55tq2cfzvdnsn5dfwfds5tftpbfvxkl5j  phase: Scheduling
vmi: testvmi4p7jpmzsmtpsjmj55tq2cfzvdnsn5dfwfds5tftpbfvxkl5j  phase: Scheduling
vmi: testvmi4p7jpmzsmtpsjmj55tq2cfzvdnsn5dfwfds5tftpbfvxkl5j  phase: Scheduling
vmi: testvmi4p7jpmzsmtpsjmj55tq2cfzvdnsn5dfwfds5tftpbfvxkl5j  phase: Scheduling
vmi: testvmi4p7jpmzsmtpsjmj55tq2cfzvdnsn5dfwfds5tftpbfvxkl5j  phase: Scheduling
vmi: testvmi4p7jpmzsmtpsjmj55tq2cfzvdnsn5dfwfds5tftpbfvxkl5j  phase: Running
datavolume: test-datavolume-thcdqz4fp6c7  phase: Succeeded
vmi: testvmi4p7jpmzsmtpsjmj55tq2cfzvdnsn5dfwfds5tftpbfvxkl5j  phase: 
vmi: testvmi4p7jpmzsmtpsjmj55tq2cfzvdnsn5dfwfds5tftpbfvxkl5j  phase: Scheduling
vmi: testvmi4p7jpmzsmtpsjmj55tq2cfzvdnsn5dfwfds5tftpbfvxkl5j  phase: Scheduling
vmi: testvmi4p7jpmzsmtpsjmj55tq2cfzvdnsn5dfwfds5tftpbfvxkl5j  phase: Scheduling
vmi: testvmi4p7jpmzsmtpsjmj55tq2cfzvdnsn5dfwfds5tftpbfvxkl5j  phase: Scheduling
vmi: testvmi4p7jpmzsmtpsjmj55tq2cfzvdnsn5dfwfds5tftpbfvxkl5j  phase: Scheduling
vmi: testvmi4p7jpmzsmtpsjmj55tq2cfzvdnsn5dfwfds5tftpbfvxkl5j  phase: Scheduling
vmi: testvmi4p7jpmzsmtpsjmj55tq2cfzvdnsn5dfwfds5tftpbfvxkl5j  phase: Scheduling
vmi: testvmi4p7jpmzsmtpsjmj55tq2cfzvdnsn5dfwfds5tftpbfvxkl5j  phase: Scheduling
vmi: testvmi4p7jpmzsmtpsjmj55tq2cfzvdnsn5dfwfds5tftpbfvxkl5j  phase: Scheduling
vmi: testvmi4p7jpmzsmtpsjmj55tq2cfzvdnsn5dfwfds5tftpbfvxkl5j  phase: Scheduling
vmi: testvmi4p7jpmzsmtpsjmj55tq2cfzvdnsn5dfwfds5tftpbfvxkl5j  phase: Scheduling
vmi: testvmi4p7jpmzsmtpsjmj55tq2cfzvdnsn5dfwfds5tftpbfvxkl5j  phase: Scheduling
vmi: testvmi4p7jpmzsmtpsjmj55tq2cfzvdnsn5dfwfds5tftpbfvxkl5j  phase: Scheduling
vmi: testvmi4p7jpmzsmtpsjmj55tq2cfzvdnsn5dfwfds5tftpbfvxkl5j  phase: Scheduling
vmi: testvmi4p7jpmzsmtpsjmj55tq2cfzvdnsn5dfwfds5tftpbfvxkl5j  phase: Running

------------------------------
• [SLOW TEST:86.142 seconds]
DataVolume Integration
/root/go/src/kubevirt.io/kubevirt/tests/datavolume_test.go:37
  Starting a VirtualMachineInstance with a DataVolume as a volume source
  /root/go/src/kubevirt.io/kubevirt/tests/datavolume_test.go:72
    using Alpine import
    /root/go/src/kubevirt.io/kubevirt/tests/datavolume_test.go:73
      should be successfully started and stopped multiple times
      /root/go/src/kubevirt.io/kubevirt/tests/datavolume_test.go:74
------------------------------
• [SLOW TEST:92.265 seconds]
DataVolume Integration
/root/go/src/kubevirt.io/kubevirt/tests/datavolume_test.go:37
  Starting a VirtualMachine with a DataVolume
  /root/go/src/kubevirt.io/kubevirt/tests/datavolume_test.go:106
    using Alpine import
    /root/go/src/kubevirt.io/kubevirt/tests/datavolume_test.go:107
      should be successfully started and stopped multiple times
      /root/go/src/kubevirt.io/kubevirt/tests/datavolume_test.go:108
------------------------------
```
</p>
</details>

mhenriks commented:
ci test please

mhenriks commented:
ci test please

cynepco3hahue commented on glide.yaml:
```diff
@@ -91,7 +91,7 @@ import:
 - package: kubevirt.io/containerized-data-importer
   repo:    https://github.com/kubevirt/containerized-data-importer.git
-  version: v1.3.0
+  version: 254749a63d335a4ec1ef919a971cab0b68cf4980
```
Why do not use the tag?

dylanzr replied:
Based on the input from @mmazur, I shifted this to a hash from a tag due to the latest tag not having a fix for api definitions.

mmazur replied:
Yes, a version with the correct api definitions makes it less of a hassle to develop python code against CDI. Hopefully 1.4.2 comes out soon, but for the time being it'd be great if after all this work what gets merged doesn't have known issues.
